### ISSUE TITLE
Cargo build error

### ISSUE THREAD

**Issue by vk496** [sjtakada/sysrepo-rs#3]
Hello,

Thank you for your great work! I tried to build your project and test a little bit sysrepo, but I found this kind of errors:

```
error[E0308]: mismatched types
    --> /var/home/vk496/.cargo/registry/src/github.com-1ecc6299db9ec823/sysrepo-0.2.0/src/lib.rs:1115:25
     |
1115 |                         options,
     |                         ^^^^^^^ expected *-ptr, found `i32`
     |
     = note: expected raw pointer `*mut *mut lyd_node`
                       found type `i32`

error[E0308]: mismatched types
    --> /var/home/vk496/.cargo/registry/src/github.com-1ecc6299db9ec823/sysrepo-0.2.0/src/lib.rs:1119:36
     |
1119 |                 Some(LydNode::from(node))
     |                                    ^^^^ expected *-ptr, found `u32`
     |
     = note: expected raw pointer `*mut lyd_node`
                       found type `u32`

error[E0308]: mismatched types
    --> /var/home/vk496/.cargo/registry/src/github.com-1ecc6299db9ec823/sysrepo-0.2.0/src/lib.rs:1129:25
     |
1129 |                         options,
     |                         ^^^^^^^ expected *-ptr, found `i32`
     |
     = note: expected raw pointer `*mut *mut lyd_node`
                       found type `i32`

error[E0308]: mismatched types
    --> /var/home/vk496/.cargo/registry/src/github.com-1ecc6299db9ec823/sysrepo-0.2.0/src/lib.rs:1133:36
     |
1133 |                 Some(LydNode::from(node))
     |                                    ^^^^ expected *-ptr, found `u32`
     |
     = note: expected raw pointer `*mut lyd_node`
                       found type `u32`

Some errors have detailed explanations: E0061, E0081, E0308, E0425.
For more information about an error, try `rustc --explain E0061`.
error: could not compile `sysrepo` due to 113 previous errors
```

I supose that this library was written for older sysrepo version (currently I have the v2.1.42). Could you please share what sysrepo version this library is intended to work?

Thank you in advance,

BR,
Valentin

**Comment by vk496** [comment]
btw, v1 builds correctly

**Comment by koalo** [comment]
Properly fixed by #4. Can be closed.
